### PR TITLE
test: Refactor test for `getCredentials` for `aws` provider

### PR DESCRIPTION
**Pull request by AlinoeDoctari**
Continue work on #8446

### CODE REVIEW

File: test/unit/lib/plugins/aws/provider.test.js [hunk 1]
```diff
@@ -690,6 +428,228 @@ describe('AwsProvider', () => {
...
+
+[notDefaultWithRole]
+source_profile = notDefault
+role_arn = NOTDEFAULTWITHROLEROLE
+`,
+        { flag: 'w+' }
+      );
+    });
+
+    after(async () => {
```
Comment (pgrzesik) [inline test/unit/lib/plugins/aws/provider.test.js:454]: This is not needed as the mocked homedir will be cleaned after each test

**Review by pgrzesik — CHANGES_REQUESTED** [review]
Thank you @AlinoeDoctari :bow: It looks great overall, I have just one small comment and we're good to go :+1:

File: test/unit/lib/plugins/aws/provider.test.js (hunk 1 above)
Comment (AlinoeDoctari) [reply test/unit/lib/plugins/aws/provider.test.js]: I was not sure :+1: thanks :bow:

**Review by pgrzesik — APPROVED** [review]
Thank you @AlinoeDoctari :bow:

**Pull request merged**